### DISCUSSION TITLE
Loss of bluetooth connection on Zwift-"Controllable"

### DISCUSSION THREAD

**Discussion by Fenar91** [doudar/SmartSpin2k#417] [Q&A]
Hi guys,
the bluetooth-connection to the zwift "controllable" isnt possible or not stable (loss of connection repeatedly) but the bluetooth connection to heartfrequency is working. I have no answer for that. Is there any possible solutions for this problem?

P.S. tried with an ipad and an android phone.

Greetings!

**Comment by doudar** [comment]
For Zwift you need to pair SS2K as a power meter first, then after a few seconds, the controllable trainer selection becomes available. This behavior is the same for all controllable trainers, not just SS2K.

If it's a stability problem (Zwift connects but then the connection goes away), a log would be helpful. You can do this via the web interface (developer tools), "debugging", or better yet over UDP by following this guide: https://github.com/doudar/SmartSpin2k/wiki/Viewing-logs-via-UDP

Good luck!

**Reply by doudar** [reply]
Also, make sure the unit has the latest firmware. We recently discovered a bug where the power meter drop down in Bluetooth scanner webpage wasn't saving the power meter selections.

**Reply by eMadman** [reply]
> Hi guys, the bluetooth-connection to the zwift "controllable" isnt possible or not stable (loss of connection repeatedly) but the bluetooth connection to heartfrequency is working. I have no answer for that. Is there any possible solutions for this problem?
> 
> P.S. tried with an ipad and an android phone.
> 
> Greetings!

There's a possibility you have a bad devkit.  I recently had to throw away 20 of them because they all failed to pass testing. 
 Thankfully, the shop on Aliexpress gave me a refund, but these stupid things have a pretty high defect rate out of the box.  Testing before soldering has saved me a lot of headache, but it's time consuming.  My current batch seems to have bad capacitors which are preventing boot from DC power without soldering an additional cap onto the devkit to supplement the stock one.
 
The recent failures had bluetoooth stability issues that were similar across all of them:

- Try to connect Zwift and it will take an abnormally long time to find the SS2K
- frequent disconnects
- difficulty connecting to SS2K via wifi while bluetooth connected (wifi and BT share a radio on these)
- Zwift connects to the power meter part of SS2K but does not detect SS2K as a cadence sensor, controllable device, or HRM - some seemingly random combination of this

**Reply by NX915** [reply]
> > Hi guys, the bluetooth-connection to the zwift "controllable" isnt possible or not stable (loss of connection repeatedly) but the bluetooth connection to heartfrequency is working. I have no answer for that. Is there any possible solutions for this problem?
> > P.S. tried with an ipad and an android phone.
> > Greetings!
> 
> There's a possibility you have a bad devkit. I recently had to throw away 20 of them because they all failed to pass testing. Thankfully, the shop on Aliexpress gave me a refund, but these stupid things have a pretty high defect rate out of the box. Testing before soldering has saved me a lot of headache, but it's time consuming. My current batch seems to have bad capacitors which are preventing boot from DC power without soldering an additional cap onto the devkit to supplement the stock one.
> 
> The recent failures had bluetoooth stability issues that were similar across all of them:
> 
>     * Try to connect Zwift and it will take an abnormally long time to find the SS2K
> 
>     * frequent disconnects
> 
>     * difficulty connecting to SS2K via wifi while bluetooth connected (wifi and BT share a radio on these)
> 
>     * Zwift connects to the power meter part of SS2K but does not detect SS2K as a cadence sensor, controllable device, or HRM - some seemingly random combination of this

Wow I've been facing those exact symptoms ever since I built mine a while back and after some hair pulling I ended up only using my SS2K for shitfting. Knowing this now, I would love to build a working one but the thought of desolder and resolder the esp32 haunts me. Do you have any suggestion as to what to look for when purchasing ESP32 boards to avoid the bad ones? And what is your suggested method to testing these before soldering? Thanks so much!

**Reply by eMadman** [reply]
Testing ESP32's is a pain in the butt - I have a test PCB that I'm using with dupont headers to pre-test before soldering them by hand.   Unfortunately, there's no rhyme or reason if you'll get a good one or a bad one.  A vendor I used for most of the past year and had many good boards from is the one that sent me 20 bad ones in one batch.  I've had other problems with ESP32's I bought off amazon. The majority also don't have a properly specced capacitor for the en pin to enable flashing without pressing buttons during the boot sequence.  It's just the pick of the crop these days.

I don't advise desoldering the ESP32 for exactly the reasons you don't want to do it.  

Email me at emadmanmakes@gmail.com and I can send you the necessary parts at cost.  Github says you're in Toronto and I'm in downtown - I can mail the parts or we can link up at a convenient location.

**Reply by Fenar91** [reply]
Hi Guys,
its crazy but i think i solved the problem with the bluetooth aka. Zwiftconection. My actual phone could not find the ssk neither in nor in zwift. 
I tried it now with my old phone, which was reseted and has almost no apps installed. He can find both ss2k, which cant be found in my actual phone, in lightspeed. And zwift ist working too with very fast connection. Is it possible, that some App on my actual phone is blocking the ss2k BLA-Signal? Im really suprised.

Greetings.

**Reply by doudar** [reply]
Great news! 

I know we used to have a lot of issues with the Zwift Companion app running in the background and blocking other apps from connecting but I think Zwift resolved that.

Possibility you have another fitness app running on your phone that is blocking the connection?

**Reply by Fenar91** [reply]
I dont really know what can issues on my phone. A lot of my friends connected succesfull and very fast too but my Phone and the one of my customer has done problems. :/

**Reply by Fenar91** [reply]
Hi Guys,
i just bought some other ESP32, which is recording to the comments a good one. And its the Espressif-Version:

https://www.amazon.de/gp/product/B08CCYWZN3/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&th=1

I will return the ones from Aliexpress. I hope this one will have a way better connection properties. 

Cheers
